src/olympia/versions/models.py

```diff
@@ -231,10 +231,15 @@ def get_url_path(self):
             return ''
         return reverse('addons.versions', args=[self.addon.slug, self.version])
 
-    def delete(self):
+    def delete(self, hard=False):
```

Comment:
I'm unsure if `force` would be a better name which is used imho quite often to describe some action that really really has to be done. (not sure if I described my intention clearly though :smile:)